bluemira/magnetostatics/greens.py

```diff
@@ -406,6 +407,57 @@ def greens_Bz(
     return 1 / x * greens_dpsi_dx(xc, zc, x, z, d_xc, d_zc)
 
 
+@nb.jit(nopython=True)
+def greens_Fz(
+    xa: Union[float, np.ndarray],
+    za: Union[float, np.ndarray],
+    xb: Union[float, np.ndarray],
+    zb: Union[float, np.ndarray],
+) -> Union[float, np.ndarray]:
+    """
+    Calculate vertical force between two coils at (xa, za) and (xb, zb) due to unit
+    currents using a Greens function.
+
+    Parameters
+    ----------
+    xa:
+        Coil x coordinates [m]
+    za:
+        Coil z coordinates [m]
+    xb:
+        Other coil x locations [m]
+    zb:
+        Other coil z locations [m]
+
+    Returns
+    -------
+    The force response from coil A to coil B [N/A^2].
+
+    Raises
+    ------
+    ZeroDivisionError:
+        If xa==xb and za==zb
+
+    Notes
+    -----
+    The absolute value of the force can be obtained by multiplying the result by
+    the Ampere-turns in both coils (i.e. Na*Ia*Nb*Ib). The force is towards coil B,
+    and is always positive. The signs of the currents then determine the direction
+    of the force. (same sign -> attractive force, opposite sign -> repulsive force).
+
+    Y. Iwasa, Case Studies in Superconducting Magnets Design and Operational Issues,
+    Second Edition, 2009, equation 3.34 and 3.36, p.83
+    ISBN 978-0-387-09799-2
+    """
+    rho = abs(za - zb)
+    factor = rho * np.sqrt((xa + xb) ** 2 + rho**2) / ((xa - xb) ** 2 + rho**2)
+    k2 = 4 * xa * xb / ((xa + xb) ** 2 + rho**2)
+    k2 = clip_nb(k2, GREENS_ZERO, 1.0 - GREENS_ZERO)
+    kk2 = ellipk_nb(k2)
+    ek2 = ellipe_nb(k2)
+    return 0.5 * MU_0 * factor * (k2 * kk2 + (k2 - 2) * (kk2 - ek2))
+
```

Comment:
@je-cook is this proof that this codecov/patch stuff is BS? I wrote a test for this guy in this same PR...

Reply:
Nope its a known Numba issue https://github.com/numba/numba/issues 4268

Reply:
We could run our tests like so `NUMBA_DISABLE_JIT=1 pytest` in order to get accurate coverage but I imagine that would be quite slow!

Reply:
Ah true... puah not sure it would be that much slower, we tend to avoid heavy lifting in tests. But not fussed either way